git-cheatsheet.md

```diff
@@ -0,0 +1,40 @@
+# git cheatsheet
+This page gathers some instructions around the git functionalities we'll use throughout the workshop, so feel free to consult at any time!
+
+## Clone
+
+For full documentation on this command check the [git docs](https://www.git-scm.com/docs/git-clone).  
+
+* To clone a repository follow the instructions on this [GitHub article](https://help.github.com/en/articles/cloning-a-repository)
+
+## Create a branch
+
+This [Git Branching doc](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging) is a very comprehensive guide to Basic branching and merging, and it's worth the read if you're new to git and it's architecture.
+Also, worth to know about [GitFlow](https://nvie.com/posts/a-successful-git-branching-model/), which is one of the most used branching models (either pure GitFlow or some variation of it).  
+
+To create a new branch:  
+
+1. `git checkout -b <your-branch-name>`
+
+## Commit
+
+For more info on commits, see [the docs](https://git-scm.com/docs/git-commit) and this useful article on [Saving changes](https://www.atlassian.com/git/tutorials/saving-changes) on git.  
+
+To commit new changes:
+1. Prepare the new files for the comming by adding them to the staging area:  
+  1.1 `git add -all`
+2. Then, commit your code via `git commit -a -m <commit-message>`  
+  2.1 Add a proper description to your commit message. Ex: Workshop1 - Adding generated react app.   
+
+## Push
+
+For more info on pushing your changes, see this [GitHub article](https://help.github.com/en/articles/pushing-to-a-remote) and [the docs](https://git-scm.com/docs/git-push)
+To push new changes:
+1. With all the changes committed, the next step is to Push the code to GitHub:    
+  1.1 `git push -u origin <your-branch-name>`  
```

Comment:
We probably need to make clear that `-u` is a one-time requirement. Git defaults nowadays mean that once you've done `git push -u origin my-cool-feature` then `git push` should suffice thereafter.